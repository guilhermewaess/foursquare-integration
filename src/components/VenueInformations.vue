<template>
  <div>
    <v-layout row>
      <v-flex xs12
              class="text-xs-center">
        <span :class="venueNameSize">{{ item.venue.name }}</span>
      </v-flex>
    </v-layout>
    <v-layout row
              class="ml-1">
      <v-flex xs12>
        <div>Address</div>
        <span class="body-1">{{ address }}</span>
      </v-flex>
    </v-layout>
  </div>
</template>

<script>
export default {
  name: 'VenueInformations',
  props: {
    item: {
      required: true,
      type: Object,
    },
  },
  computed: {
    address() {
      return this.item.venue.location.formattedAddress.toString().replace(/,/gi, ', ');
    },
    venueNameSize() {
      if (this.item.venue.name.length > 30) {
        return 'headline';
      }
      return 'display-1';
    },
  },
};
</script>

<style lang="scss" scoped>
.headline{
  line-height: 40px !important;
}
</style>
