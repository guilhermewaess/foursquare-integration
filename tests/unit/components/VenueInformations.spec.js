import { shallowMount } from '@vue/test-utils';
import VenueInformations from '@/components/VenueInformations.vue';

describe('VenueInformations', () => {
  let item;
  let component;
  beforeEach(() => {
    item = {
      venue: {
        name: 'Venue',
        location: {
          formattedAddress: ['Aleja Genera 39', '30-001 KrakÃ³w', 'Poland'],
        },
      },
    };
    component = shallowMount(VenueInformations, {
      propsData: { item },
      stubs: ['v-layout', 'v-flex'],
    });
  });
  it('should have name', () => {
    expect(component.name()).toEqual('VenueInformations');
  });
  describe('snapshot', () => {
    it('should render', () => {
      expect(component.element).toMatchSnapshot();
    });
  });
  describe('computed', () => {
    it('address should formatted address', () => {
      const expected = item.venue.location.formattedAddress.toString().replace(/,/gi, ', ');
      expect(component.vm.address).toEqual(expected);
    });
    it('venueNameSize should return display-1 if name length is at least 30 ', () => {
      expect(component.vm.venueNameSize).toEqual('display-1');
    });
    it('venueNameSize should return headline if name length more than 30 ', () => {
      item.venue.name = 'VenueWithMoreThan30CharsForTests';
      component.setProps({ item: Object.assign({}, item) });

      expect(component.vm.venueNameSize).toEqual('headline');
    });
  });
});
