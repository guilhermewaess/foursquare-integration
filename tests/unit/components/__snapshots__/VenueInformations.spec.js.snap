// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VenueInformations snapshot should render 1`] = `
<div>
  <v-layout-stub
    row=""
  >
    <v-flex-stub
      class="text-xs-center"
      xs12=""
    >
      <span
        class="display-1"
      >
        Venue
      </span>
    </v-flex-stub>
  </v-layout-stub>
   
  <v-layout-stub
    class="ml-1"
    row=""
  >
    <v-flex-stub
      xs12=""
    >
      <div>
        Address
      </div>
       
      <span
        class="body-1"
      >
        Aleja Genera 39, 30-001 Krak√≥w, Poland
      </span>
    </v-flex-stub>
  </v-layout-stub>
</div>
`;
